src/foam/nanos/blob/FileService.java

```diff
@@ -49,8 +56,33 @@ protected void download(X x) {
         return;
       }
 
+      // parse query parameters
```

Comment:
I think it would be better to put session support in the AuthenticatedWebAgent, that way it would be reusable.

Reply:
Yeah you're right. I've moved it to AuthWebAgent